File: problems/Contiguous Array/findMaxLength.cs
public class Solution {
    public int FindMaxLength(int[] nums) {
        var result = 0;
        var count = 0;
        var map = new Dictionary<int, int> {{0, -1}};

        for (var i = 0; nums.Length > i; ++i) {
            count += (nums[i] == 1 ? 1 : -1);

            if (map.ContainsKey(count)) {
                result = Math.Max(result, i - map[count]);
            } else {
                map.Add(count, i);
            }
        }

        return result;
    }
}

File: problems/Contiguous Array/findMaxLength.java
class Solution {
    public int findMaxLength(int[] nums) {
        int result = 0;
        int count = 0;
        HashMap<Integer, Integer> map = new HashMap<>() {{put(0, -1);}};

        for (int idx = 0; nums.length > idx; ++idx) {
            count += (1 == nums[idx] ? 1 : -1);

            if (map.containsKey(count)) {
                result = Math.max(result, idx - map.get(count));
            } else {
                map.put(count, idx);
            }
        }

        return result;
    }
}
